### ISSUE TITLE
how to use cameras' known intrinsics for SFM?

### ISSUE THREAD

**Issue by squashking** [openMVG/openMVG#1416]
Hi, I have 4 images taken by 4 cameras and each camera's intrinsics are known and different. Can I use the known instrinsics to improve 3D reconstruction? I see there is an optional parameter -k for openMVG_main_SfMInit_ImageListing, but it seems I can only specify k for only one camera, any methods to specify k for all the 4 cameras? Thank you!

**Comment by pmoulon** [comment]
I would advise you to create your own tiny initializer (starting from SfMInit_ImageListing).
You can create your 4 camera model here (camera intrinsics) and then link the images to the right camera intrinsic ID.

**Comment by squashking** [comment]
@pmoulon Thank you! So what I need to do is write a sfm_data.json file specifying intrinsics in the file for each camera, am I right? is there any sample code for writing sfm_data.json?

**Comment by squashking** [comment]
@pmoulon update: I have found sample code in main_SfMInit_ImageListing.cpp, thanks.

**Comment by pmoulon** [comment]
Yes, you can write all the thing you need in the sfm_data.json and then call Incremental SfM2 or ComputeStructureFromKnowPoses to use your camera poses and just perform the triangulation and bundle adjustment step.

**Comment by squashking** [comment]
thank you!
